### PR TITLE
Improve module documentation

### PR DESCRIPTION
**Pull request by ayeshLK**
Fixes: [#6088](https://github.com/ballerina-platform/ballerina-library/issues/6088)

## Examples
N/A

## Checklist
- [X] Linked to an issue
- [ ] ~Updated the changelog~
- [ ] ~Added tests~
- [ ] ~Updated the spec~
- [ ] ~Checked native-image compatibility~

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,105 +5,127 @@
 [![Trivy](https://github.com/ballerina-platform/module-ballerinax-ibm.ibmmq/actions/workflows/trivy-scan.yml/badge.svg)](https://github.com/ballerina-platform/module-ballerinax-ibm.ibmmq/actions/workflows/trivy-scan.yml)
 [![GitHub Last Commit](https://img.shields.io/github/last-commit/ballerina-platform/module-ballerinax-ibm.ibmmq.svg)](https://github.com/ballerina-platform/module-ballerinax-ibm.ibmmq/commits/main)
```
Comment (RDPerera) [inline README.md:6]: Is it possible to add a GraalVM batch as well? (with manually dispatchable workflow)

File: README.md [hunk 2]
```diff
@@ -5,105 +5,127 @@
...
 
-The `ballerinax/ibm.ibmmq` library provides an API to connect to an IBM MQ server using Ballerina.
+[IBM MQ](https://www.ibm.com/products/mq) is a powerful messaging middleware platform designed for facilitating reliable
+communication between disparate systems and applications. IBM MQ ensures the secure and orderly exchange of messages
+asynchronously, decoupling senders and receivers for efficient and scalable communication. It supports both
+point-to-point and publish/subscribe messaging models via queues and topics.
 
-This library is created with minimal deviation from the IBM MQ java client API to make it easy for the developers who are used to working with the IBM MQ java client.
+The `ballerinax/ibm.ibmmq` package provides an API to connect to an IBM MQ server using Ballerina.
```
Comment (RDPerera) [inline README.md:13]: Is there any particular version/code name of the IBM MQ side we support? If yes, It'd be great to include that information :) ?

File: README.md [hunk 3]
```diff
@@ -5,105 +5,127 @@
...
+#### Produce message to an IBM MQ queue
 
 ```ballerina
-import ballerinax/ibm.ibmmq;
+check queue->put({
+    payload: "This is a sample message to IBM MQ queue".toBytes()
+});
+```
+
+#### Produce message to an IBM MQ queue
```
Comment (dilanSachi) [inline README.md:105]:
```suggestion
#### Produce message to an IBM MQ topic
```

File: README.md [hunk 4]
```diff
@@ -5,105 +5,127 @@
...
+#### Retrieve messages from an IBM MQ topic
+
+```ballerina
+ibmmq:Message? message = check topic->get();
 ```
 
+### Examples
+
+The following example shows how to use the `ibm.ibmmq` connector to produce and consume messages using an IBM MQ server.
+
```
Comment (dilanSachi) [inline README.md:128]: is the example missing?

File: README.md [hunk 5]
```diff
@@ -5,105 +5,127 @@
...
+![Click on Manage](https://raw.githubusercontent.com/ballerina-platform/module-ballerinax-ibm.ibmmq/main/docs/setup/resources/click-manage-link.png)
+
+5. Navigate to `Events` tab.
+
+![Navigate to Events tab](https://raw.githubusercontent.com/ballerina-platform/module-ballerinax-ibm.ibmmq/main/docs/setup/resources/navigate-to-events-tab.png)
+
+6. Click on `Create`.
+
+![Click on create](https://raw.githubusercontent.com/ballerina-platform/module-ballerinax-ibm.ibmmq/main/docs/setup/resources/click-on-create.png)
```
Comment (dilanSachi) [inline README.md:43]: shall we crop the screenshots to remove the menu bar in the browser?

File: README.md (hunk 1 above)
Comment (ayeshLK) [reply README.md:6]: We have not added the graalvm check for this connector.

File: README.md (hunk 5 above)
Comment (ayeshLK) [reply README.md:43]: Fixed with [77ff4e23d6c4be861035a1dd16cf133cfba89d19](https://github.com/ballerina-platform/module-ballerinax-ibm.ibmmq/pull/37/commits/77ff4e23d6c4be861035a1dd16cf133cfba89d19)

File: README.md (hunk 3 above)
Comment (ayeshLK) [reply README.md]: Fixed with [aec20e2d75706fbcbb9cb3d9ca01e87e920cd1bf](https://github.com/ballerina-platform/module-ballerinax-ibm.ibmmq/pull/37/commits/aec20e2d75706fbcbb9cb3d9ca01e87e920cd1bf)

File: README.md (hunk 2 above)
Comment (ayeshLK) [reply README.md:13]: Fixed with [3d2e8413695f1c4aded7605c26c918cc6314ba76](https://github.com/ballerina-platform/module-ballerinax-ibm.ibmmq/pull/37/commits/3d2e8413695f1c4aded7605c26c918cc6314ba76)

File: README.md [hunk 6]
```diff
@@ -5,105 +5,134 @@
...
+
+## Quickstart
+
+To use the IBM MQ connector in your Ballerina application, modify the .bal file as follows:
+
+### Step 1: Import the connector
+
+Import `ballerinax/ibm.ibmmq` package into your Ballerina project.
```
Comment (NipunaRanasinghe) [inline README.md:58]:
```suggestion
Import `ballerinax/ibm.ibmmq` module into your Ballerina project.
```

**Review by dilanSachi — APPROVED** [review]
LGTM

File: README.md (hunk 6 above)
Comment (ayeshLK) [reply README.md]: Fixed with [5b2525eb3fcbccd00e96b0a9c5cc12994bc3b9c6](https://github.com/ballerina-platform/module-ballerinax-ibm.ibmmq/pull/37/commits/5b2525eb3fcbccd00e96b0a9c5cc12994bc3b9c6)

**Review by NipunaRanasinghe — APPROVED** [review]
LGTM

**Pull request merged**